From 31df353eeb62dd9a9a7b902604146612492122a6 Mon Sep 17 00:00:00 2001
From: Xianzhong <xianzhong.li@nxp.com>
Date: Tue, 23 Oct 2018 01:28:57 +0800
Subject: [PATCH 4909/5242] MGS-4338-5 [#imx-1217] remove duplicate query code
 for debugfs

commit  20a5878345cd65b3f8a7dc72bc1e7408c33c2fb4 from
https://source.codeaurora.org/external/imx/linux-imx.git

remove the duplicate code with gckGALDEVICE_QueryFrequency,
gckHARDWARE_QueryFrequency has correct fix for power management.

Signed-off-by: Xianzhong <xianzhong.li@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../hal/os/linux/kernel/gc_hal_kernel_device.c     |  104 +-------------------
 1 file changed, 1 insertion(+), 103 deletions(-)

diff --git a/drivers/mxc/gpu-viv/hal/os/linux/kernel/gc_hal_kernel_device.c b/drivers/mxc/gpu-viv/hal/os/linux/kernel/gc_hal_kernel_device.c
index 2fd668a..8f8657f 100644
--- a/drivers/mxc/gpu-viv/hal/os/linux/kernel/gc_hal_kernel_device.c
+++ b/drivers/mxc/gpu-viv/hal/os/linux/kernel/gc_hal_kernel_device.c
@@ -70,10 +70,6 @@
 
 extern gcTA globalTA[16];
 
-gceSTATUS
-gckGALDEVICE_QueryFrequency( IN gckGALDEVICE Device);
-
-
 /******************************************************************************\
 ******************************** Debugfs Support *******************************
 \******************************************************************************/
@@ -702,8 +698,6 @@ static int gc_clk_show(struct seq_file* m, void* data)
     gckGALDEVICE device = node->device;
     gctUINT i;
 
-    gckGALDEVICE_QueryFrequency(device);
-
     for (i = gcvCORE_MAJOR; i < gcvCORE_COUNT; i++)
     {
         if (device->kernels[i])
@@ -716,6 +710,7 @@ static int gc_clk_show(struct seq_file* m, void* data)
                 continue;
             }
 #endif
+            gckHARDWARE_QueryFrequency(hardware);
 
             if (hardware->mcClk)
             {
@@ -2099,103 +2094,6 @@ static irqreturn_t isrRoutineVG(int irq, void *ctxt)
 
 /*******************************************************************************
 **
-**  gckGALDEVICE_QueryFrequency
-**
-**  Query frequency for all the hardwares.
-**
-*/
-gceSTATUS
-gckGALDEVICE_QueryFrequency(
-    IN gckGALDEVICE Device
-    )
-{
-    gctUINT64 mcStart[gcvCORE_COUNT] = { 0 }, shStart[gcvCORE_COUNT];
-    gctUINT32 mcClk[gcvCORE_COUNT], shClk[gcvCORE_COUNT];
-    gckHARDWARE hardware = gcvNULL;
-    gceSTATUS status;
-    gctUINT i;
-
-    gcmkHEADER_ARG("Device=0x%p", Device);
-
-    for (i = gcvCORE_MAJOR; i < gcvCORE_COUNT; i++)
-    {
-#if gcdENABLE_VG
-        if (i == gcvCORE_VG)
-        {
-            continue;
-        }
-#endif
-
-        if (Device->kernels[i])
-        {
-            hardware = Device->kernels[i]->hardware;
-
-            mcStart[i] = shStart[i] = 0;
-
-            if (Device->args.powerManagement)
-            {
-                gcmkONERROR(gckHARDWARE_SetPowerManagement(
-                    hardware, gcvFALSE
-                    ));
-            }
-
-            gcmkONERROR(gckHARDWARE_SetPowerManagementState(
-                hardware, gcvPOWER_ON_AUTO
-                ));
-
-            gckHARDWARE_EnterQueryClock(hardware,
-                                        &mcStart[i], &shStart[i]);
-        }
-    }
-
-    gcmkONERROR(gckOS_Delay(Device->os, 50));
-
-    for (i = gcvCORE_MAJOR; i < gcvCORE_COUNT; i++)
-    {
-        mcClk[i] = shClk[i] = 0;
-
-#if gcdENABLE_VG
-        if (i == gcvCORE_VG)
-        {
-            continue;
-        }
-#endif
-
-        if (Device->kernels[i])
-        {
-            hardware = Device->kernels[i]->hardware;
-
-            if (mcStart[i])
-            {
-                gckHARDWARE_ExitQueryClock(hardware,
-                                           mcStart[i], shStart[i],
-                                           &mcClk[i], &shClk[i]);
-            }
-
-            hardware->mcClk = mcClk[i];
-            hardware->shClk = shClk[i];
-
-            if (Device->args.powerManagement)
-            {
-                gcmkONERROR(gckHARDWARE_SetPowerManagement(
-                    hardware, gcvTRUE
-                    ));
-            }
-        }
-    }
-
-    gcmkFOOTER_NO();
-
-    return gcvSTATUS_OK;
-
-OnError:
-    gcmkFOOTER();
-
-    return status;
-}
-
-/*******************************************************************************
-**
 **  gckGALDEVICE_Start
 **
 **  Start the gal device, including the following actions: setup the isr routine
-- 
1.7.9.5

